### DISCUSSION TITLE
Sprint 14

### DISCUSSION THREAD

**Discussion by Justus-at-Tazama** [tazama-lf/General-Issues#317] [General]
## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

### @cshezi - Cebolenkosi Shezi

**Yesterday:**

**Today:**

### @vorsterk - Kyle Vorster

**Yesterday:**

**Today:**

### @Lenbkr - Len Bekker

**Yesterday:**

**Today:**

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:**

**Today:**

### @rtkay123 - Rodney Kanjala

**Yesterday:**

**Today:**

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

**Comment by Justus-at-Tazama** [comment]
## 2023/11/13

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Reingestion of PoC data. No missed transactions. Some transactions missing from the Sybrin reporting. Started with the defaulting of values on the Sybrin Configuration Manager. Found a library (Fox Services) that can ease integration with Arango.

**Today:** Check in on missing transactions from the reporting.

### @vorsterk - Kyle Vorster

**Yesterday:** Redis benchmarking for a bit, then paused to create some reporting tools to check the key sizes of the various artefacts in Redis. Working on the Rust typology processor - the processor is going through several iterations. The Rust processor keeps on restarting the pods and we are not currently sure why. Jason is going to assist with troubleshooting. So far performance gains looks good, but the restarts are hampering the evaluation.

**Today:** Carrying on with the bash file for redis. The script pauses intermittently which is strange. Then moving on to the Rust typology processor. Will then use the Jupyter report to see results.

### @Lenbkr - Len Bekker

**Yesterday:** Demo preparation.

**Today:** Demo preparation.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo preparation.

**Today:** Demo preparation.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Working on the event sidecar for logging. Updated the protofile to include services where we define the capabilities of the event sidecar. Integrated the sidecar into a new service we created to batch up the logs before sending them to Elastic. Working on the Rust typology processor. Reworked the typology processor to make most of the startup components configurable.

**Today:** Clarifying the requirement for log batching prior to elastic feed, then implement. Rust typology processor with Kyle and Jason.

### @johanfol - Johan Foley

**Yesterday:** 

**Today:** Check in on PoC. Plan for the demo.

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/11/14

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Sybrin Configuration Manager Stuff (Logic) Remote SB, Wrote a document on Foxx Services of Arango to discuss the use cases that can be covered by these services from eKuta's business logic.

**Today:** Continuing with SCM and anything else that comes up in eKuta side like the patching of the batch recently done

### @vorsterk - Kyle Vorster

**Yesterday:** Demo prep , Did some research into redis insight , installed that locally . Got the average TADP key sizes and the rest . Ran some benchmarks with 1.2 million requests for each key size. My local was fine but when I tested on SB1 the vms maxed out.

**Today:** Hoping to complete the redis benchmarking today on dev and monitoring vms to check usage and throttle points.

### @Lenbkr - Len Bekker

**Yesterday:** Demo prep

**Today:** Demo prep

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo prep

**Today:** Demo prep

### @rtkay123 - Rodney Kanjala

**Yesterday:** Centralized logging/sidecar implementation

**Today:** Centralized logging/sidecar implementation

### @johanfol - Johan Foley

**Yesterday:**

**Today:** PoC reporting update

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/10/09

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Patched the batch ingestion into the reporting environment. Had to update the script. Continued with the Sybrin Configuration Manager.

**Today:** New ticket.

### @vorsterk - Kyle Vorster

**Yesterday:** Redis benchmarking. Ran Redis across 3 VMs and collected data. Noted some inconsistencies in the deletion of keys. Saw high CPU utilisation on one of the VMs compared to the others. There are a couple of opportunities to improve the performance.

**Today:** Updating SB3. Exporting Jenkins jobs from SB1. Justus to support with e2e testing.

### @Lenbkr - Len Bekker

**Yesterday:** Started updating SB3. Demo prep.

**Today:** demo prep - SB3 update delegated to Kyle.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo prep

**Today:** Demo prep

### @rtkay123 - Rodney Kanjala

**Yesterday:** Spinning up local elastic to test the submission of logging through to Kibana. Started working on the integration into the library for the centralisation of services for use across the processes.

**Today:** Continue with the lumberjack sidecar and centralised logging.

### @johanfol - Johan Foley

**Yesterday:** 

**Today:** Reporting integration update.

## Parking lot

New surprise ticket.

SB3 deployment - Kyle to take over and deploy Main; Sybrin configuration manager and reporting to be deployed after EKUTA deployment. SCM to be deployed from SB1 snapshot - anticipating about a day's work on the Sybrin side.

**Comment by Justus-at-Tazama** [comment]
## 2023/11/16

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Reviewed the requirements for the optimisation of the typology processor's interaction with Redis. Ticket ready to go. The rules list is now being created when the ruleResults is received. Able to create the typologyResult from the list. 

**Today:** Perform the typology completion evaluation. And garbage collection.

### @vorsterk - Kyle Vorster

**Yesterday:** Redeployed SB3 with the latest main branch. Picked up an issue with the CRSP, typology and TADProc deployments. It initially appeared to be an Arango config issue. Turned out it was hard-coded values in those deployment jobs. Updated the parameters and completed the deployment.

**Today:** Assisting Rodney with the event sidecar deployment. Will work on the Typology Processor by creating a new container for the sidecar parallel to the typology processor. Redeploy for testing onto SB1. Revisit Redis analytics for review by Jason.

### @Lenbkr - Len Bekker

**Yesterday:** Demo prep - final stages

**Today:** Demo prep - integrating feedback

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo prep - final stages

**Today:** Demo prep - integrating feedback

### @rtkay123 - Rodney Kanjala

**Yesterday:** Working on the event sidecar for logging. Updated the deployment YAML files. Also started working on the library. Proposing to start on the TMS API for the event sidecar instead of the typology processor.

**Today:** Test and deploy!

### @johanfol - Johan Foley

**Yesterday:** Design of the typology processor optimisation. Complete reporting.

**Today:** Complete the reporting. There may be an issue in the CMS service that is not working optimally. Then we need to rerun the ingestion.

## Parking lot

Use the local Sybrin server for a performance benchmarking "staging" or "pre-prod" environment to save time and costs. The idea is to do troubleshooting where it's cheaper. The official performance benchmarks will always be conducted in the cloud.

Rust processor needs some additional work - some collaboration around the troubleshooting required.

Typology mystery: in the event of a rule being submitted to the typology processor twice, the rule result should be upserted and not inserted to prevent duplicates. We should be using the `addOneGetAll` function going forward and not the `addOneGetCount` function. Once a single ruleResult has been posted, the ENTIRE array of rules that had previously been submitted must be returned to the typology processor for processing for the individual typologies.

**Comment by Justus-at-Tazama** [comment]
## 2023/11/17

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Working in updating the typology processor to the new optimised flow. Issues with the flow.

**Today:** Continue with the typology processor. Unit testing.

### @vorsterk - Kyle Vorster

**Yesterday:** Worked with Rodney on the deployment and service YAMLs on the event sidecar.  Much work, many sweat - it is now working. Working on Micro-Kubernetes on local servers in preparation for local benchmarking and documentation for local on-premises deployments. Deleted the sb4 cluster and cleaned up repo images for sb4 as well in the registry. Shut down SB4.

**Today:** Pursue micro-kubernetes (mk8s), deployment and testing.

### @Lenbkr - Len Bekker

**Yesterday:** Demo prep

**Today:** Demo prep

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo prep. Looked into the Go issue for the typology processor.

**Today:** Demo prep. 

### @rtkay123 - Rodney Kanjala

**Yesterday:** Event sidecar. Managed to get all the processor pieces connected. Even sidecar is talking to the new logger. Was able to see logs in Elastic, but only the setup logs.

**Today:** Request and conduct a code review before integrating into Main.

### @johanfol - Johan Foley

**Yesterday:** Sybrin reporting for PoC.

**Today:** Travelling

## Parking lot

Sybrin deployments.

**Comment by Justus-at-Tazama** [comment]
## 2023/11/20

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Working on updating the typology processor flow. Completed the interpretation of the rules list by each of the typologies. On Friday results were being duplicated after the typology processor. Found and fixed this duplication issue. Evaluating multiple transactions at a time were working fine. Moved on to unit testing. Recommend that we test the branch before integration into Main. Suggesting a batch test. Foley suggested completing the ticket entirely before we do any performance benchmarking.

**Today:** Complete unit tests.

### @vorsterk - Kyle Vorster

**Yesterday:** Assisting Rodney with the event sidecar on SB1. Working on Mk8s - authentication was resolved.

**Today:** Complete mk8s deployment and run some baseline benchmarks to see what the basic performance is - then deploying Shezi's updated typology processor if it's ready. Standing by to support Rodney if anything is required.

### @Lenbkr - Len Bekker

**Yesterday:** Demo prep. Updated the Sybrin reporting environment onto SB3. Looked at the config manager displaying only a single typology processor. Did not want to bring the whole environment down yet until work complete.

**Today:** Demo feedback updates. (Perhaps) SCM on SB3 if JP is ready.

Status: Ekuta e2e is up and running. Reporting environment is up and running. SCM is outstanding until confirmed ready to go.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo prep.

**Today:** Catch up on the issue on the typology processor. Sybrin config manager with Len. Demo feedback updates.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Event sidecar on Friday with Kyle. Logs are appearing in Elastic when services are starting up. Operational logs (e.g. TMS API) are not logging through yet. Logs are going to the logging service, but not posting. Suspect it's a log-level thing. Startup logs are trace-levels.

**Today:** Get the operational logs up and running.

### @johanfol - Johan Foley

**Yesterday:** Demo stuff.

**Today:** Assist where required. Logging, Typology Processor. Start presentation for Matt review session this evening.

## Parking lot

**Comment by rtkay123** [comment]
## 2023/11/21

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Justus-Ortlepp-frms 

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Optimising TP, completing unit tests, nearly done with closing the ticket, just testing now looking out for performance. Saw opportunity to optimise transactions and be clever about sending smaller (only needed) data instead of the full payload. Changes not yet included in development branch, just a theory.

**Today:** Benchmark TP with Kyle

### @vorsterk - Kyle Vorster

**Yesterday:** Busy setting up local servers, got everything installed, Arango and indexes. Worked with @rtkay123 on his elastic issues. Might be an issue with his log indexes.

**Today:** Going to run a benchmark and run @cshezi stuff. Work with Rodney to fix indexes on Kibana. Empty logs may be the cause

### @Lenbkr - Len Bekker

**Yesterday:** SB3 nitro environment backup. Noticed some breaking changes in terms of config and payload. Everything works but Sybrin to Arango not tested yet. No errors but no information is showing sometimes.

**Today:** Check if happy path is working. Run past @JeanPierreNell  if not. Fix microflows that reference new fields. Documentation touch up on updating configs

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Checked out some issues with @Lenbkr, he helped point out the issues. Managed to implement most of the features from the provided feedback in Demo prep. Basically done

**Today:** Prepare and schedule for demo prep meeting.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Worked on lumberjack logging. Issues with certs and indexes. @johanfol helped resolve most of them. Still no logs in elastic, but no cert errors. Further debugging needed, indexes may be the cause

**Today:** work with @vorsterk for resolving index issues on elastic.

### @johanfol - Johan Foley

**Yesterday:** Helped @rtkay123 look at cert issues and integration with pino and elastic. Took a look at @cshezi TP changes

**Today:** Further look at TP changes, assist where needed

**Comment by JeanPierreNell** [comment]
## 2023/11/22

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Tried to create case test for minimisation of the transaction object. Foley called in for sit-down for PR-Review. Found a few more improvements, found a way to check typologies without using redis. Pulled numbers down to the exact number of rules. Continued with Rodney's review for a few more optimisations. Removal of some foreach loops. Tested optimisations, Rodney pulled changes into Rust. Looked into diffrent ticket - calculation of execution time. Discussed it with Foley, Foley paused it.

**Today:** Looking into a new Ticket.

### @vorsterk - Kyle Vorster

**Yesterday:** Working with Rodney in the morning to get the sidecar stuff working. Paused that, Rodney will continue by himself for now. Issues was raised on github related to our problems. Busy with local testing, e2e wasn't working. Investigated, grabed config from SB1. Issue was when SB4 was deleted, a shadow resource group was deleted along with it, this impacted Dev, cluster will have to be recreated. Send E-Mail to X-Content for assistance. DevOps can't assist. Looked into exporting information from persistent volumes. Jenkins / Arango config still missing. Worked a little on Teraform. Read up on Locks for resource groups.

**Today:** If X-Content is unable to assist, will work on terraform in azure, since the data can be used and imported. Hopefully can get e2e working on local Sybrin Servers. About 20 times faster.  

### @Lenbkr - Len Bekker

**Yesterday:** Finalised Demo

**Today:** Check in with Jp about the Configuration Manager.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Finalising Demo. Looked at config manager

**Today:** SCM - Config Manager

### @rtkay123 - Rodney Kanjala

**Yesterday:** Testing a bunch of ways to make a logging service display logs in elastic. Kyle assisted a bit. Tried diffrent versions of elastic. No errors, elastic spinning up from docker doesn't have all the configurations. Jumped in a call with Shezi to look at the PR - noticed nested for loop. Managed to speed it up quite a bit.

**Today:** Investigate Data steams. Try to get Elastic working. 

### @johanfol - Johan Foley

**Yesterday:** 

**Today:**

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/11/23

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Completed the documentation for the typology processor optimization. Processor deployed for QA and tested. Submitted for performance benchmarking. Started working on the ALRT status in the channel result.

**Today:** ALRT status on channel result should be done today. And then a new ticket following.

### @vorsterk - Kyle Vorster

**Yesterday:** Assisting Rodney with the event sidecar. Created new indexes and created a new data view. There appears to be an issue in Pino. Replacing a Jenkins resource group that was deleted when we closed down SB4. Trying to get the resource group restored by Azure to restore the persistent volumes. Testing end-to-end on the local deployment in Sybrin. Main branch is not currently working in dev - try to coordinate with the current deployment state in SB1.

**Today:** Get e2e up on the server blades and run some benchmarks. Waiting for Azure to get back on the restore.

### @Lenbkr - Len Bekker

**Yesterday:** Working on deploying the SB3 Sybrin config manager. Updated the micro-flows. The resynch should work.

**Today:** Align with JP and test the UI.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Configuration manager and added differentiating stuff for the rules that were missing.

**Today:** Alignment and e2e testing with Len. Working on building the exception for Rule 078 (cased results).

### @rtkay123 - Rodney Kanjala

**Yesterday:** Investigating empty logs in Elastic (log entry exists, but there's no content). Found a Pino DSK to send to a custom network socket. Constructed a network transport to interface with logstash directly, but still has the same issue. There are some open issues on the Pino side that we are monitoring. There are some environment variables and Node version issues that we can try if we're desperate. Pino has a wrapper around elastic that handles the transmission and we expect it should be formatted correctly.

**Today:** Figure out what to do to make the sidecar work. Foley suggests making the flushbyte setting to 1 to force an immediate transmission. Check in with Kyle on pino indexing.

### @johanfol - Johan Foley

**Yesterday:**

**Today:** BCEAO presentation.

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/11/24

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Starting the new tickets to build the routing based on the network map. Will be working on a parallel branch until after BCEAO. Completed the documentation update of the new typology processor flow.

**Today:** If the typology processor benchmarking is done today, we can merge the changes.

### @vorsterk - Kyle Vorster

**Yesterday:** Assisting Rodney with the event sidecar. Eventually managed to rename the mapping in Elastic and message contents were being displayed. Working on rolling back the message mapping earlier in the flow. Created another job to clean up the disk space issue. Testing the recreation of a new volume. End to end up and running on performance benchmarking - typology expressions were out of date in the configuration. Ran about 1500 transactions, but getting a very high error rate. The transactions do not appear to be accepted by the TMS API.

**Today:** Debug the ingestion for the performance benchmarking.

### @Lenbkr - Len Bekker

**Yesterday:** Internal Sybrin support. Completed the Sybrin Configuration Manager - waiting for final sign-off to send the documentation to JoPacc. Heads-up on Sybrin license expiry next week.

**Today:** Documentation and wrap up of the SCM.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:**

**Today:**

### @rtkay123 - Rodney Kanjala

**Yesterday:** Working on getting the logging contact onto Kibana. Data mapping issue resolved. The transport is not respecting the format in which the message is distributed. Trying to resolve the mapping.

**Today:** Rolling out the event sidecar to the remaining processors.

### @johanfol - Johan Foley

**Yesterday:** Preparing for the demo sessions next week.

**Today:** Wrap up demo prep.

## Parking lot

**Comment by vorsterk** [comment]
## 2023/11/27

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** TP new ticket , routing using the network map. Completed TP work. Justus was assisting with ironing out details and made changes. Create branch and waiting till Foley and Justus.

**Today:** TADP work like TP work completed. 

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on figuring out why transactions are getting lost when submitting through jmeter during benchmark testing. No errors and everything seems fine !!! 

**Today:** Carry on figuring out why im losing transactions when benchmarking on local servers.

### @Lenbkr - Len Bekker

**Yesterday:** Last second issues for Sybrin config manager , transaction config not being populated anywhere???? 

**Today:** Look into Sybrin stuff. 

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Last second issues for Sybrin config manager  , took Foley through it . Call with Roland for demo related stuff .

**Today:** Documentation for SCM , on our bookshelf.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Rolling out logging changes to the rest of the processors . Revert if the side car isn't available to the old logging . Duplicate code removing.

**Today:** Finishing up the logging changes and Lib and testing it on a branch.

### @johanfol - Johan Foley

**Yesterday:** Demo

**Today:** Demo

## Parking lot

None

**Comment by JeanPierreNell** [comment]
## 2023/11/28

## Attendees

@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Continued with ticket to create NATS routing using the networkmap from the TP side. Need some clearance when it comes to messages. 

**Today:** Needs Testing. New Ticket.

### @vorsterk - Kyle Vorster

**Yesterday:** Working on getting the benchmarks up on the local Sybrin Servers. Arango Index had the incorrect name and kept failing. Ran a benchmark which worked. Spoke with Daniel, we only have 2 VMs on our locals, scale up as much as we can, but won't reach the numbers dev reached. 

**Today:** Testing on Local, will run benchmarks on Shezi's Code. Test Rodney's Rust as well. If happy, will run on dev. Still speaking with the Azure guys for a call. They could maybe restore our persistent volumes. 

### @Lenbkr - Len Bekker

**Yesterday:** Sybrin Internal Things

**Today:** Sybrin Internal Things

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Finished documentation on SCM. 

**Today:** Working on GO Processor, waiting for feedback from Foley.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Finished rolling out the logging changes. Mainly worked on the library. Added some documentation, refactoring in the processor side-car (No testing yet, but should be good, YOLO) 

**Today:** Need Code review. Work on Documentation. Testing , need to change indexes. Maybe include GRPC client with the rust, MAYBE.

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

## Parking lot

**Comment by cshezi** [comment]
## 2023/11/29

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Took Len's ticket for the base fixture of the unit test, Working on the central test code repository, foreseeing a slowdown due to many unit tests that have to go through. Linking PRs to tickets, facing a backlog of around four tickets waiting for team members to return. completed for Typology Processor yesterday

**Today:** Continuing with the ticket of implementing the base fixtures for our unit test

### @vorsterk - Kyle Vorster

**Yesterday:** I reached out to Azure regarding persistent volumes, provided them a detailed breakdown via email, and waiting for their response. Conducted benchmarking on the main branch, aiming for 100 transactions/second; currently achieving around 69 transactions/second after optimizing Redis instances from 64 to 12 due to pod limitations on VMs. Similarly reduced Nats connections from 12 to 3 for optimization.

**Today:** Testing Typology processor breeds, Rust, Go, and Node for the Redis Improvement changes

### @Lenbkr - Len Bekker

**Yesterday:** Internal Sybrin Stuff Node Red...

**Today:** Internal Sybrin Stuff Node Red...

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Worked on the GO Typology Processor to accommodate the new changes implemented on the Node version

**Today:** Continuing with GO TP with the half day I have

### @rtkay123 - Rodney Kanjala

**Yesterday:** I realized I hadn't addressed changing the default logger to use Pino when the sidecar wasn't configured. I implemented this using log4js and tested it locally, but not end-to-end due to complexities in setting up the local stack, particularly with Elastic. Started drafting documentation for the upcoming merge to main, focusing on explaining the logging framework changes. Clarified questions with Foley and asked Shezi for code review, planning to involve Len and JP for input.

**Today:** I'll focus on the final code review for the logging changes and setting up GRPC clients for Rust.

### @johanfol - Johan Foley

**Yesterday:** 

**Today:**

## Parking lot

**Comment by rtkay123** [comment]
## 2023/11/30

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Justus-Ortlepp-frms 

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** 
Continued with base fixture ticket for unit tests, sorted out typology-processor, started with tadproc, incomplete yet, sorting out today and moving on with other tickets. 

**Today:** 
Possibly merging to main if branches aren't locked and continuing with the ticket (https://github.com/frmscoe/General-Issues/issues/123)

### @vorsterk - Kyle Vorster

**Yesterday:** 
Running some benchmarks, increased performance from Redis branches (~46%) from Sybrin local servers. Testing on Dev pending... Thoughts to scale up TADP, as it may bottleneck newer TPs

**Today:** 
Carry on with perf benchmarks, testing on `Dev environment`. Environment setup with SQL databases, going to consult @rtkay123 if hurdles are encountered. Meeting with Devops stuff, documentation as focal point

### @Lenbkr - Len Bekker

**Yesterday:** Found outdated nodes(plugins) for Node Red, not working properly, incomplete setup for now. 

**Today:** 
Testing JetStream node ~ finding something that works with NATS.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** 
Continued with Go processor, made two branches, will be pushed soon. Sending to Kyle afterwards. Documentation on config manager

**Today:** 
Finishing processor and QA

### @rtkay123 - Rodney Kanjala

**Yesterday:** 
Requesting Code Review and some refactors on lib.


**Today:** 
Code Review with returning members of team. Further testing of new branches since sandboxes aren't locked anymore. Helping @vorsterk with SQL issues if they come up

### @johanfol - Johan Foley

**Yesterday:** 

**Today:** Admin stuff

## Parking lot

**Comment by JeanPierreNell** [comment]
## 2023/10/09

## Attendees
@johanfol - Johan Foley
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Started with a new ticket. Merged the improvement to Typology processor. Ticket is merged, comments from Foley was resolved. New Ticket is the removal of the status on the channel level. Seems like a major update - breaking the proto buff. Seems like there is a way to remove the attribute without breaking the encoding - reserving the tag. 

**Today:** Looking to merge the new ticket. Will continue with using routing from networkmap.

### @vorsterk - Kyle Vorster

**Yesterday:** Working on local Sybrin Servers, optomized it. Benchmarking on dev with the new typology changes. With the disks lost for arango, they need to be recreated - but there is some issues with creation. Reached out to Sybrin DevOps. Call with cross lakes guy - they gave sum good pointers and extra stuff we can add in. His name is Bill.

**Today:** Disk Creation. As soon as Arango disk is created, a test can be run. 

### @Lenbkr - Len Bekker

**Yesterday:** On Leave

**Today:** On Leave

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Working on GO processor

**Today:** Finish up GO processor.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Testing on Sandboxes for the new logging framework. Waiting before breaking the main library. Sacrificed a little bit of code readability in favour of backwards compatibility. Looked at elastic data views. 

**Today:** Creating diffrent coloms for diffrent fields in kibana. Call with JP 

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

## Parking lot
